.github/workflows/ci.yml

```
@@ -0,0 +1,459 @@
+          if [[ "${{ matrix.db }}" =~ mysql ]]; then
+            if [ "${{ matrix.db }}" == "mysql57pdo" ]; then
+              cat << EOF > .env
+          SS_DATABASE_CLASS="MySQLPDODatabase"
+          SS_DATABASE_PORT="3357"
+          EOF
+            else
+              cat << EOF > .env
+          SS_DATABASE_CLASS="MySQLDatabase"
+          SS_DATABASE_PORT="${{ matrix.db == 'mysql57' && '3357' || '3380' }}"
+          EOF
+            fi
+            cat << EOF >> .env
+          SS_DATABASE_SERVER="127.0.0.1"
+          SS_DATABASE_USERNAME="root"
+          SS_DATABASE_PASSWORD="root"
+          EOF
+          else
+            cat << EOF > .env
+          SS_DATABASE_CLASS="PostgreSQLDatabase"
+          SS_DATABASE_SERVER="localhost"
+          SS_DATABASE_PORT="5432"
+          SS_DATABASE_USERNAME="postgres"
+          SS_DATABASE_PASSWORD="postgres"
+          EOF
+          fi
+          cat << EOF >> .env
+          SS_ENVIRONMENT_TYPE="dev"
+          SS_DATABASE_NAME="SS_mysite"
+          SS_DEFAULT_ADMIN_USERNAME="admin"
+          SS_DEFAULT_ADMIN_PASSWORD="password"
+          SS_TRUSTED_PROXY_IPS="*"
+          SS_MFA_SECRET_KEY="1234567894175b99966561e1efe237e4"
+          SS_BASE_URL="http://localhost"
+          EOF
+
+          # debug
+          echo ".env is"
+          cat .env
+
+          # Artifacts directory must be created after composer install as it would remove the artifacts directory
+          mkdir artifacts
+
+          # run dev/build flush to help debug any issues (though it's not strictly required here)
+          # travis-shared has no reference to sake, so it should be safe to omit dev/build flush=1
+          # normal module
+          if [ -f vendor/bin/sake ]; then
+            vendor/bin/sake dev/build flush=1
+          fi
+          # framework module
+          if [ -f sake ]; then
+            ./sake dev/build flush=1
+          fi
+
+          # Delete the silverstripe-cache dir - it will automatically recreate when needed
+          # There were issues with a unit test getting the following issue
+          # Identifier name 'SilverStripe_CampaignAdmin_Tests_AddToCampaignValidatorTest_TestObject' is too long
+          # Likely because the /tmp/silverstripe-cache-php7.4.xyz... dir being out of sync with TestOnly objects
+          rm -rf $(find /tmp -maxdepth 1 | grep silverstripe-cache)
+
+      - name: Debug
+        run: |
+          echo "matrix.phpunit: ${{ matrix.phpunit }}"
+
+      - name: Run tests
+        uses: emteknetnz/gha-run-tests@main
+        with:
+          phpunit: ${{ matrix.phpunit }}
+          phpunit_suite: ${{ matrix.phpunit_suite }}
+          endtoend: ${{ matrix.endtoend }}
+          endtoend_suite: ${{ matrix.endtoend_suite }}
+          endtoend_config: ${{ matrix.endtoend_config }}
+          js: ${{ matrix.js }}
+          phplinting: ${{ matrix.phplinting }}
+          phpcoverage: ${{ matrix.phpcoverage }}
```

```suggestion
          endtoend: ${{ matrix.endtoend }}
          endtoend_suite: ${{ matrix.endtoend_suite }}
          endtoend_config: ${{ matrix.endtoend_config }}
          phpcoverage: ${{ matrix.phpcoverage }}
          phplinting: ${{ matrix.phplinting }}
          phpunit: ${{ matrix.phpunit }}
          phpunit_suite: ${{ matrix.phpunit_suite }}
          js: ${{ matrix.js }}
```
Rearranged to match the order of inputs in the action itself.
Why don't we pass in `phpunit_fail_on_warning`? If that input isn't used, should it be removed from the action?

---

I don't like how it looks reordered :-) Have added in `phpunit_fail_on_warning: false`